sympy/core/add.py

```diff
@@ -859,10 +859,33 @@ def _eval_as_leading_term(self, x):
 
         infinite = [t for t in expr.args if t.is_infinite]
 
-        expr = expr.func(*[t.as_leading_term(x) for t in expr.args]).removeO()
-        if not expr:
-            # simple leading term analysis gave us 0 but we have to send
-            # back a term, so compute the leading term (via series)
+        compute = False
+        leading_terms = [t.as_leading_term(x) for t in expr.args]
+        new_expr = expr.func(*leading_terms).removeO()
```

Comment:
It is important that only leading terms of lowest order are included here. For example, if the expression is `1/(1 - x) - 1 + x**2`, then the leading terms are `[1, -1, x**2]` and their sum `x**2` is non-zero. However, the series of the expression is `x + 2*x**2 + ...`, which shows that the correct leading term is `x`. That can be found if `new_expr` is computed as the sum of the lowest order leading terms `[1, -1]`.